physics/Interstitials/UFS_SCM_NEPTUNE/GFS_phys_time_vary.scm.F90

```diff
@@ -769,21 +769,15 @@ subroutine GFS_phys_time_vary_timestep_init (
          rjday = jdoy + jdat(5) / 24.
          if (rjday < ozphys%time(1)) rjday = rjday + 365.
 
-         n2 = ozphys%ntime + 1
-         do j=2,ozphys%ntime
-            if (rjday < ozphys%time(j)) then
-               n2 = j
-                      exit
-            endif
-         enddo
-         n1 = n2 - 1
-         if (n2 > ozphys%ntime) n2 = n2 - ozphys%ntime
+         call find_photochemistry_index(ozphys%ntime, ozphys%time, rjday, n1, n2)
 
 !> - Update ozone concentration.
          if (ntoz > 0) then
             call ozphys%update_o3prog(jindx1_o3, jindx2_o3, ddy_o3, rjday, n1, n2, ozpl)
          endif
 
+         call find_photochemistry_index(h2ophys%ntime, h2ophys%time, rjday, n1, n2)
```

Comment:
this line should move inside if (h2o_phys) then?

Reply:
I moved all calls inside the relevant conditional, does that work?